### PR TITLE
Feature/sa remove fileuploader

### PR DESCRIPTION
**Pull request by sarimabbas**
Removal of FileUploader and transition to regular file upload using firebase. 
Also made EditProjectImages component independent, which allows for image edits in real time

### CODE REVIEW

**Comment by bpalomino5** [comment]
Screen Shot 2019-04-06 at 6 32 08 PM
EditProject: Does not load up any image

**Comment by sarimabbas** [comment]
Which project did you try this on? 

I think for existing projects, it may be failing because it is looking in a different directory structure in Cloud Storage.

File: src/components/submit-project/SubmitProject.jsx [hunk 1]
```diff
@@ -107,10 +56,6 @@ class SubmitProject extends Component {
       card_des,
       links,
       need,
-      isUploading,
-      avatarURL,
-      progress,
-      images,
       category } = this.state;
     return (
       <React.Fragment>
```
Comment (bpalomino5) [inline src/components/submit-project/SubmitProject.jsx]: You can use new react fragment style `<> </>`

File: src/components/submit-project/components/edit-project-images/EditProjectImages.jsx [hunk 2]
```diff
@@ -1,66 +1,99 @@
...
+
+  // the state
   const [ loading, setLoading ] = useState(false);
   const [ imageFiles, setImageFiles ] = useState([]);
   const [ imagePreviews, setImagePreviews ] = useState([]);
-  //   useEffect(() => console.log(imagePreviews, imageFiles))
 
+  // retrieve existing images on mount
+  useEffect(() => {
+    handleExistingImages()
```
Comment (bpalomino5) [inline src/components/submit-project/components/edit-project-images/EditProjectImages.jsx]: Effect is being called before being defined, move function up

File: src/components/submit-project/components/edit-project-images/EditProjectImages.jsx [hunk 3]
```diff
@@ -1,66 +1,99 @@
...
+    setLoading(false);
   };
 
+  /**
+   * React component which is used to implement the drag-drop behavior
+   * wraps each image thumbnail
+   */
   const SortableImagePreview = SortableElement(({ preview, index }) => (
     <div className={styles.image}>
       <img src={preview} className={styles.imageDOM} />
```
Comment (bpalomino5) [inline src/components/submit-project/components/edit-project-images/EditProjectImages.jsx]: Either use Image from reactstrap or handle all default img props i.e. src, alt, className etc.

File: src/components/submit-project/components/edit-project-images/EditProjectImages.jsx [hunk 4]
```diff
@@ -84,12 +121,15 @@ const EditProjectImages = (props) => {
       ))}
       {/* add image input */}
       <div className={styles.placeholder}>
-        <div className={styles.placeholderItem} onClick={handleInputClick}>
+        {/* your own image */}
+        <div className={styles.placeholderItem} onClick={handleInputClick} role="input">
```
Comment (bpalomino5) [inline src/components/submit-project/components/edit-project-images/EditProjectImages.jsx]: Need to handle all the clickable divs. `eslint(jsx-a11y/click-events-have-key-events)`
Either use a Button component via reactstrap, handle the key event, or ignore this eslint issue if you are confident about doing so.

Please handle all the clickable divs.

File: src/components/submit-project/components/edit-project-images/EditProjectImages.jsx [hunk 5]
```diff
@@ -101,24 +141,32 @@ const EditProjectImages = (props) => {
...
 
+  /**
+   * when a user uploads a file on their computer
+   * @param {*} event : on click
+   */
   const handleAddImage = async (event) => {
     const { target: { files } } = event;
     const file = files[0];
     setLoading(true);
+    console.log([ file, ...imageFiles ])
```
Comment (bpalomino5) [inline src/components/submit-project/components/edit-project-images/EditProjectImages.jsx]: no console logs please, only use for debugging and errors (as of right now)

File: src/components/submit-project/SubmitProject.jsx [hunk 6]
```diff
@@ -127,24 +72,6 @@ class SubmitProject extends Component {
               onChange={event => this.setState({ contact: event.target.value })}
             />
           </FormGroup>
-          <EditProjectImages handleUploadImages={this.handleUploadImages} />
```
Comment (bpalomino5) [inline src/components/submit-project/SubmitProject.jsx]: I can't upload files on submit project form? how can i test this feature?

**Comment by bpalomino5** [comment]
> Which project did you try this on?
> 
> I think for existing projects, it may be failing because it is looking in a different directory structure in Cloud Storage.

Tried it on Thinkspaces project. I have edit access there. My test project was deleted awhile ago.

**Comment by bpalomino5** [comment]
Screen Shot 2019-04-06 at 6 34 42 PM
Clean up compile warnings

File: src/components/submit-project/SubmitProject.jsx (hunk 6 above)
Comment (sarimabbas) [reply src/components/submit-project/SubmitProject.jsx]: Currently, you should be able to test it by creating a new project, and then editing it. In a later commit, I removed `<EditProjectImages/>` from `SubmitProject.jsx` entirely, the idea being to put most of the editing features in `EditProject.jsx`. This is something I am still designing mockups for.

**Comment by sarimabbas** [comment]
> > Which project did you try this on?
> > I think for existing projects, it may be failing because it is looking in a different directory structure in Cloud Storage.
> 
> Tried it on Thinkspaces project. I have edit access there. My test project was deleted awhile ago.

I wonder how feasible it is to migrate old projects to the new directory structure? 

The old ones use: storage > projectpictures > *
The new ones use: storage > images > projects > pid > listing > *

**Comment by bpalomino5** [comment]
> > > Which project did you try this on?
> > > I think for existing projects, it may be failing because it is looking in a different directory structure in Cloud Storage.
> > 
> > 
> > Tried it on Thinkspaces project. I have edit access there. My test project was deleted awhile ago.
> 
> I wonder how feasible it is to migrate old projects to the new directory structure?
> 
> The old ones use: storage > projectpictures > *
> The new ones use: storage > images > projects > pid > listing > *

I suggest doing a migration script/function in order to transfer all the data to appropriate directories

File: src/components/submit-project/SubmitProject.jsx (hunk 6 above)
Comment (bpalomino5) [reply src/components/submit-project/SubmitProject.jsx]: Screen Shot 2019-04-08 at 10 09 30 PM
First Error

Comment (bpalomino5) [reply src/components/submit-project/SubmitProject.jsx]: Screen Shot 2019-04-08 at 10 11 48 PM
Refreshed the page, tried again, and got this.

Comment (bpalomino5) [reply src/components/submit-project/SubmitProject.jsx]: I also noticed that upon saving the changes the images do not show up on carousel until a page refresh?

**Comment by sarimabbas** [comment]
Sounds good, I can get started on a one-time Cloud Function that should be able to connect to the database and make the migration

**Comment by sarimabbas** [comment]
Migration of project images is complete. All project images now live inside `Cloud Storage > images > project > pid > listing > *`

Also patched a bug that arises from clicking Save button.

File: src/components/submit-project/SubmitProject.jsx (hunk 6 above)
Comment (sarimabbas) [reply src/components/submit-project/SubmitProject.jsx]: Ahh, I believe `ProjectPage` tries to save an entire project object to the database when the Save button is clicked. Unfortunately, this state is out of sync with the actual project images after you use the `EditProjectImages` picker. 

I fixed this in a later comment below by triggering a re-mount on save.

**Pull request merged**